src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -904,6 +904,12 @@ bool tryProcessTree(SyntaxTree partialTree, [NotNullWhen(true)] out (AnalysisSco
                     {
                         compilationEvents = compilationEventsForTree.AddRange(compilationEvents);
 
+                        // Filter out synthesized compilation unit completed event that was generated for span analysis
+                        // as we are now doing full tree analysis, GetCompilationEventsForSingleFileAnalysis call above should
+                        // have already generated a CompilationUnitCompletedEvent without any filter span.
+                        compilationEvents = compilationEvents.WhereAsArray(e => e is not CompilationUnitCompletedEvent c || !c.FilterSpan.HasValue);
```

Comment:
We were generating duplicate CompilationUnitCompletedEvents for SymbolStart/End analyzers, which was identified by the added test.